### ISSUE TITLE
ReflectionClassLoader on Java 16

### ISSUE THREAD

**Issue by jokbon** [lucko/LuckPerms#2853]
**invalid plugin exception**
Log paste: https://bytebin.lucko.me/RnZpFdicbT (modified to remove unsafe stuff)
Reproduced on both **paper** and **yaptopia** (**bukkit**).

luckerms is the only plugin that doesn't work on java 16
There are plenty of reasons to use java 16
java 16 gets 36cps, java 15 just 27 (on my server)
its way faster and it fixes garbage colection bugs from previous versions.

**Comment by lucko** [comment]
Thanks for reporting, I'll look into this and get back to you :)

**Comment by BomBardyGamer** [comment]
According to the new Java 9 module system, this shouldn't even work with Java 9, due to the [much more strict encapsulation that was introduced that also applies to reflection](https://stackoverflow.com/questions/41265266/how-to-solve-inaccessibleobjectexception-unable-to-make-member-accessible-m/41265267)

**Comment by lucko** [comment]
It works fine up until J16 because we're able to open the package to the 'unnamed module' at runtime.

Seems this is no longer possible, so will need a new solution.

**Comment by jokbon** [comment]
Shud i use pidgeon upstream? Its in english? No breaking changes? @karlatemp 
是英文吗？没有不兼容的更改？
https://github.com/Karlatemp/LuckPerms-Mirai/commit/45ea8dab5bc37acf4f515d6b13dc70bfe1b7fd57
bump, upstream might contain valuable code

**Comment by lucko** [comment]
Hey, I think I'm going to try a more long term solution instead of more hacks - but thank you!

**Comment by lucko** [comment]
```
➜  /Users/luck/Desktop/servers/tuinity /Library/Java/JavaVirtualMachines/zulu-16.0.83.jdk/bin/java -DPaper.IgnoreJavaVersion=true -jar server.jar nogui
Unsupported Java detected (60.0). Only up to Java 15 is supported.
System Info: Java 16 (OpenJDK 64-Bit Server VM 16-ea+33) Host: Mac OS X 11.1 (aarch64)
Loading libraries, please wait...
...
[18:23:13 INFO]: This server is running Tuinity version git-Tuinity-"d3ee222" (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
...
[18:23:13 INFO]: [LuckPerms] Loading LuckPerms v5.2.107
[18:23:13 INFO]: Server permissions file permissions.yml is empty, ignoring it
[18:23:13 INFO]: [LuckPerms] Enabling LuckPerms v5.2.107
[18:23:13 INFO]:         __
[18:23:13 INFO]:   |    |__)   LuckPerms v5.2.107
[18:23:13 INFO]:   |___ |      Running on Bukkit - Tuinity
[18:23:13 INFO]:
[18:23:13 INFO]: [LuckPerms] Loading configuration...
[18:23:13 INFO]: [LuckPerms] Loading storage provider... [H2]
[18:23:13 INFO]: [LuckPerms] Loading internal permission managers...
[18:23:13 INFO]: [LuckPerms] Performing initial data load...
[18:23:14 INFO]: [LuckPerms] Successfully enabled. (took 449ms)
```

I have a solution that works nicely on J16 without any hacks :)

It needs quite a bit of tidying up, but I'll push it as soon as that's done.

**Comment by lucko** [comment]
Fixed in 75d8f00400cd521b7931476aa60f7ae364ed2bcd
